src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ('addProperties'):
+        Object.assign(stateCopy, action.extraData);
+        resultArr.push({ ...stateCopy });
```

Comment:
you do this push in every cases, just move it down and do it in after switch case